File: app/src/main/java/echo/com/surveys/SurveyApplication.java
package echo.com.surveys;

import android.app.Application;
import echo.com.surveys.dependencyInjection.DIModule;
import echo.com.surveys.dependencyInjection.DIComponent;
import echo.com.surveys.dependencyInjection.DaggerDIComponent;

public class SurveyApplication extends Application {

    private static SurveyApplication instance;
    private DIComponent component;
    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        component = DaggerDIComponent.builder().dIModule(new DIModule(getApplicationContext()))
                .build();
    }
    public static SurveyApplication getInstance() {
        return instance;
    }
    public DIComponent component() {
        return component;
    }
}

File: app/src/main/java/echo/com/surveys/SurveyApplication.kt
package echo.com.surveys

import android.app.Application
import echo.com.surveys.dependencyInjection.DIComponent
import echo.com.surveys.dependencyInjection.DIModule
import echo.com.surveys.dependencyInjection.DaggerDIComponent

class SurveyApplication : Application() {
    private var component: DIComponent? = null
    override fun onCreate() {
        super.onCreate()
        instance = this
        component = DaggerDIComponent.builder().dIModule(DIModule(applicationContext))
            .build()
    }

    fun component(): DIComponent? {
        return component
    }

    companion object {
        var instance: SurveyApplication? = null
            private set
    }
}
